File: zxing-sample/src/main/java/me/dm7/barcodescanner/zxing/sample/BaseScannerActivity.java
//package me.dm7.barcodescanner.zxing.sample;
//
//import androidx.appcompat.app.ActionBar;
//import androidx.appcompat.app.AppCompatActivity;
//import androidx.appcompat.widget.Toolbar;
//
//import android.view.MenuItem;
//
//public class BaseScannerActivity extends AppCompatActivity {
//
//    public void setupToolbar(Toolbar toolbar) {
//        setSupportActionBar(toolbar);
//        final ActionBar ab = getSupportActionBar();
//        if (ab != null) {
//            ab.setDisplayHomeAsUpEnabled(true);
//        }
//    }
//
//    @Override
//    public boolean onOptionsItemSelected(MenuItem item) {
//        // Respond to the action bar's Up/Home button
//        if (item.getItemId() == android.R.id.home) {
//            finish();
//            return true;
//        }
//        return super.onOptionsItemSelected(item);
//    }
//}

File: zxing-sample/src/main/java/me/dm7/barcodescanner/zxing/sample/BaseScannerActivity.kt
package me.dm7.barcodescanner.zxing.sample

import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar

open class BaseScannerActivity : AppCompatActivity() {

    fun setupToolbar(toolbar: Toolbar) {
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (item.itemId == android.R.id.home) {
            finish()
            true
        } else {
            super.onOptionsItemSelected(item)
        }
    }
}
